forge/src/main/java/org/openjdk/skara/forge/PullRequestUtils.java

```diff
@@ -128,11 +128,8 @@ private static Hash findSourceHash(PullRequest pr, Repository localRepo, List<Co
         var sourceHead = fetchMergeSource(pr, localRepo);
 
         // Ensure that the source and the target are related
-        try {
-            localRepo.mergeBase(targetHash(localRepo), sourceHead);
-        } catch (IOException e) {
-            throw new CommitFailure("The target and the source branches do not share common history - cannot merge them.");
-        }
+        localRepo.mergeBaseOptional(targetHash(localRepo), sourceHead)
+                .orElseThrow(() -> new CommitFailure("The target and the source branches do not share common history - cannot merge them."));
```

Comment:
Here I think a real error should be let through so the whole WorkItem may be retried later.